lib/engine/game/base.rb

```diff
@@ -905,6 +905,20 @@ def add_extra_tile(tile)
         extra_cities.each { |c| @_cities[c.id] = c }
       end
 
+      def share_ability_money_on_par(corporation)
+        return unless corporation.capitalization == :incremental
+
+        @companies.each do |company|
+          next unless (ability = company.abilities(:shares))
```

Comment:
let's use the block syntax

companies.abilities(:shares) do |ability|
 ...
end

Reply:
I don't think that works but **all_companies_with_ability** does.